revokePermission.ps1

```diff
@@ -0,0 +1,283 @@
+    }
+    catch {
+        $ex = $PSItem
+        $errorMessage = Get-ErrorMessage -ErrorObject $ex
+
+        Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+
+        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                # Action  = "" # Optional
+                Message = "Error creating authorization headers. Error Message: $($errorMessage.AuditErrorMessage)"
+                IsError = $true
+            })
+
+        # Throw terminal error
+        throw
+    }
+
+    # Get current account
+    try {
+        Write-Verbose "Querying account with id [$($actionContext.References.Account.Id)]"
+        $splatWebRequest = @{
+            Uri             = "$baseUrl/scim/users/$($actionContext.References.Account.Id)"
+            Headers         = $headers
+            Method          = 'GET'
+            ContentType     = "application/json;charset=utf-8"
+            UseBasicParsing = $true
+        }
+
+        $currentAccount = Invoke-RestMethod @splatWebRequest -Verbose:$false
+
+        if (($currentAccount | Measure-Object).count -gt 1) {
+            throw "Multiple accounts found with id [$($actionContext.References.Account.Id)]. Please correct this so the accounts are unique."
+        }
+        elseif (($currentAccount | Measure-Object).count -eq 0) {
+            throw "No account found with id [$($actionContext.References.Account.Id)]."
+        }
+    }
+    catch {
+        $ex = $PSItem
+        $errorMessage = Get-ErrorMessage -ErrorObject $ex
+    
+        Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+        Write-Verbose "URI: $($splatWebRequest.Uri)"
+
+        if ($errorMessage.AuditErrorMessage -Like "*No account found*" -or $errorMessage.AuditErrorMessage -Like "*(404) Not Found.*" -or $errorMessage.AuditErrorMessage -Like "*User not found*") {
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    # Action  = "" # Optional
+                    Message = "Skipped revoking permission for account with id [$($actionContext.References.Account.Id)]. Reason: No longer exists"
+                    IsError = $false
+                })
+        }
+        else {
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    # Action  = "" # Optional
+                    Message = "Error querying account with id [$($actionContext.References.Account.Id)]. Error Message: $($errorMessage.AuditErrorMessage)"
+                    IsError = $true
+                })
+
+            # Throw terminal error
+            throw
+        }
+    }
+
+    # Revoke permission
+    try {
+        Write-Verbose "Revoking permission to $($pRef.Name) ($($pRef.id)) for $($currentAccount.userName) ($($currentAccount.id))"
```

Comment:
$pRef should be in $actionContext. This goes for all the other references in code below.

Reply:
Correct, as been changed accordingly.